internal/provider/resource_ger.go

```diff
@@ -0,0 +1,191 @@
+package provider
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/squadcast/terraform-provider-squadcast/internal/api"
+	"github.com/squadcast/terraform-provider-squadcast/internal/tf"
+)
+
+func resourceGER() *schema.Resource {
+	return &schema.Resource{
+		Description: "Global Event Rules aims to simplify the process of configuring alerting in Squadcast by allowing users to create global rulesets for routing alerts to the correct service.",
+
+		CreateContext: resourceGERCreate,
+		ReadContext:   resourceGERRead,
+		UpdateContext: resourceGERUpdate,
+		DeleteContext: resourceGERDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceGERImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "GER id.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"team_id": {
+				Description:  "Team id.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: tf.ValidateObjectID,
+			},
+			"name": {
+				Description:  "GER name.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"description": {
+				Description: "GER description.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"routing_key": {
+				Description: "GER routing key.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"entity_owner": {
+				Description: "GER owner.",
+				Type:        schema.TypeList,
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Description:  "GER owner id.",
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: tf.ValidateObjectID,
+						},
+						"type": {
+							Description:  "GER owner type. (user or squad or team)",
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"user", "squad", "team"}, false),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceGERImport(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	d.SetId(d.Id())
+
```

Comment:
Is this tested? don't we need to fetch the resource details using getByID API before setting the resource?

Reply:
This is tested, we just need to set the ID here and rest will be handled in the Read function